mirror.go

```diff
@@ -48,8 +48,16 @@ func (l logWriter) Write(p []byte) (n int, err error) {
 	return len(p), nil
 }
 
+type DockerClient interface {
+	Info() (*docker.DockerInfo, error)
+	TagImage(string, docker.TagImageOptions) error
+	PullImage(docker.PullImageOptions, docker.AuthConfiguration) error
+	PushImage(docker.PushImageOptions, docker.AuthConfiguration) error
+	RemoveImage(string) error
+}
+
 type mirror struct {
-	dockerClient *docker.Client  // docker client used to pull, tag and push images
+	dockerClient DockerClient    // docker client used to pull, tag and push images
```

Comment:
I guess we weren't able to figure out why this can't be a pointer?

Reply:
It can, but then we have to reference it oddly, via parenthesis in the call functions, added more in slack on this one